checkprovider

```diff
@@ -191,6 +205,13 @@ for ( my $test = 0 ; !$period || $test < $total ; $total && $test++ ) {
     # for child processes
     if ($pid) {
 
+        # advance the term list position for the next child
+        if ( ($position + 1) > $#terms ) {
```

Comment:
It took me a minute to realize that this was a means to loop back around since this program is period & rate based instead of based on the number of elements in the list :). I kept thinking that there should be a `last` in here, but :+1: all good.